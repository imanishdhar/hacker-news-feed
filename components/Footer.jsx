import React from 'react';
import Link from 'next/link';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faAngleDoubleRight,
  faAngleDoubleLeft,
} from '@fortawesome/free-solid-svg-icons';

const Footer = ({ page }) => (
  <footer className="fixed-footer">
    {page > 1 && (
      <Link href="/news/[pageid]" as={`/news/${page - 1}`}>
        <a className="gap">
          <FontAwesomeIcon icon={faAngleDoubleLeft} />
        </a>
      </Link>
    )}

    <Link href="/news/[pageid]" as={`/news/${page + 1}`}>
      <a className="gap">
        <FontAwesomeIcon icon={faAngleDoubleRight} />
      </a>
    </Link>

    <style jsx>{`
      a {
        text-decoration: none;
      }

      a:hover {
        opacity: 0.6;
      }

      .gap {
        margin: 8px;
        font-size: 24px;
        color: #000000;
      }

      .fixed-footer {
        width: 100%;
        left: 0;
        position: fixed;
        background: #ff6600;
        padding: 10px 0;
        color: #fff;
      }

      .fixed-footer {
        bottom: 0;
        text-align: center;
      }
    `}</style>
  </footer>
);

export default Footer;
