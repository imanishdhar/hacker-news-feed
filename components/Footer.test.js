import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

function setup() {
  const props = {
    page: 1,
  };
  const wrapper = shallow(<Footer></Footer>);
  return { wrapper, props };
}

describe('Footer Test Suite', () => {
  it('Should have a more link', () => {
    const { wrapper } = setup();
    expect(wrapper.find('a').exists()).toBe(true);
  });
  it('Footer component should render', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
