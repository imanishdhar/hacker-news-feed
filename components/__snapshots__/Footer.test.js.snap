// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Test Suite Footer component should render 1`] = `
Object {
  "props": Object {
    "page": 1,
  },
  "wrapper": ShallowWrapper {},
}
`;
